src/TabNavigator.re

```diff
@@ -56,26 +56,39 @@ module type TabConfig = {
   type tabs;
   type order = list(tabs);
   let order: order;
+  let containerName: string;
   let tabBarOptions: tabBarOptions;
   let getTab:
-    tabs => (Js.Dict.key, navigation => ReasonReact.reactElement, screenOptions);
+    (tabs, navigation) =>
+    (Js.Dict.key, unit => ReasonReact.reactElement, screenOptions);
 };
 
 module Create = (Config: TabConfig) => {
   [@bs.deriving abstract]
   type navigatorConfig = {initialRouteName: string};
-  
 
   [@bs.deriving abstract]
   type routeConfig = {
-    screen: navigation => ReasonReact.reactElement,
+    screen: unit => ReasonReact.reactElement,
     navigationOptions: screenOptions,
   };
 
+  module NavigationProp = {
+    [@bs.send] external navigate: string => unit = "navigate";
+
+    [@bs.send] external goBack: unit => unit = "goBack";
+  };
+
+  let makeNavigationProp = () => {
+    navigate: routeName => NavigationProp.navigate(routeName),
+    goBack: () => NavigationProp.goBack(),
+  };
+
   let tabs =
     Config.order
     |> List.map(tab => {
-         let (tabname, screen, screenOptionsConfig) = Config.getTab(tab);
+         let (tabname, screen, screenOptionsConfig) =
+           Config.getTab(tab, makeNavigationProp());
```

Comment:
`makeNavigationProp` in the future will accept original navigation prop. How do we handle that case? I know that the status of navigation prop is in WIP but I'm wondering how we handle this

Reply:
I wonder that too 😉 We need to discuss that. You, me and @grabbou

Reply:
I think we should handle that just like we handle it in StackNavigator -> https://github.com/callstackincubator/bs-react-navigation/blob/aaf0f84250e35e49bcb0c0ee0281aa038f271bc7/src/StackNavigator.re#L55-L63